lisa_commands/request_helper.py

```diff
@@ -203,6 +208,75 @@ def gify_search(media_type, query):
 
     return None
 
+def spotify_search(query):
+    """
+    submit a search to spotify
+    :query what type of content we are querying for and the search terms themselves
+    """
+
+    TOKEN_ENDPOINT = 'https://accounts.spotify.com/api/token'
+    SEARCH_ENDPOINT = 'https://api.spotify.com/v1/search'
+    QUERY_TYPES = ['track', 'album', 'artist', 'playlist']
+
+    query_parts = query.split()
+
+    # Pop first part of query off list, this will be the query type
+    query_type = query_parts.pop(0)
+
+    if query_type not in QUERY_TYPES:
+        return 'Invalid Media Type for search. Valid types are "track", "album", "artist" or "playlist"'
+    
+    # The parts of the list that are left will be the search terms
+    search_terms = " ".join(query_parts)
+
+    # First we need to get an access token to do the search request
+
+    token_request = urllib.request.Request(TOKEN_ENDPOINT)
+
+    # Spotify expects an authorization header in the format Base64(client_id:client_secret)
+    authorization = base64.b64encode(bytes(SPOTIFY_CLIENT_ID+':'+SPOTIFY_CLIENT_SECRET, encoding='utf8')).decode('utf-8')
+    
+    token_request.add_header(
+        'Content-Type',
+        'application/x-www-form-urlencoded'
+    )
+    token_request.add_header(
+        'Authorization',
+        f'Basic {authorization}'
+    )
+    token_params = urllib.parse.urlencode({
+        'grant_type': 'client_credentials'
+    }).encode('utf-8')
+
+
+    token_response = urllib.request.urlopen(token_request, data=token_params)
+
+    # OPTIONAL TODO: Could store this token in some sort of cache (Redis or whatever equivalent AWS product) so that we don't need to 
```

Comment:
Went ahead and created a new issue for this https://github.com/Splyth/slack_bot/issues/30